.gitlab-ci.yml

```diff
@@ -67,9 +67,8 @@ check:nix-dry:
   needs: []
   script:
     - >
-      nix-shell --arg ci true --run $'
-      npmDepsHash="$(prefetch-npm-deps ./package-lock.json)";
-      nix-build -v -v --dry-run ./release.nix --argstr npmDepsHash "$npmDepsHash";
+      nix develop .#ci --command bash -c $'
+      ls -l /tmp
       '
```

Comment:
Oh you are not using `nix-shell`.

Ok well like I said, you can do these 2:

```
check:nix-dry:
  stage: check
  needs: []
  script:
    - stat /tmp
    - ls -l /tmp
```

And

```
check:nix-dry:
  stage: check
  needs: []
  script:
    - >
      nix develop .#ci --command bash -c $'
      stat /tmp;
      ls -l /tmp;
      '
```

Also can you try not using `--command bash` I think this could be adding something weird into the environment...

Reply:
![image](https://github.com/MatrixAI/Polykey-CLI/assets/640797/0420ce27-d8a5-45e2-9803-d1e21d06efb8)

Reply:
Use `nix run` instead?

Reply:
Oh and:

> --impure: Allows accessing non-Nix-provided dependencies and resources from the environment, bypassing the typical purity guarantees of Nix. Useful when the development environment needs access to system libraries or tools not specified in the flake.

You can use this for `nix develop`.

Reply:
And apparently you can do `nix develop -c 'echo Hello';`. It just runs that within a script...

Reply:
`nix run` is different, it will run an `app` defined inside of the flake. So in this scenario, you could do something like `nix run .#polykey-cli` and it would build and run Polykey-CLI, without you needing to go to the build outputs and run it yourself. You cannot just do `nix run 'nix develop ...'` or whatnot, its only for app outputs explicitly defined in the `flake.nix`.

The definition in code:
```
        apps = {
          polykey-cli = {
            type = "app";
            program = "${self.packages.${system}.polykey-cli}/bin/polykey";
          };
        };
```